### ISSUE TITLE
looking forward the training code

### ISSUE THREAD

**Issue by syf0518** [chrisdcs/LAMA-Learned-Alternating-Minimization-Algorithm#1]

**Comment by syf0518** [comment]
Thank you for your great work.I modified your training code for joint sparse-view reconstruction and metal artifact reduction in CT images, but I found a problem while training the network.As you can see in the figure below, epochs keep looping between 13 and 14, I think training is in an endless cycle now.
![屏幕截图 2023-11-28 115112](https://github.com/chrisdcs/LAMA-Learned-Alternating-Minimization-Algorithm/assets/104410659/ff89c6d9-a7a0-4550-851e-e62262b8817a)

**Comment by chrisdcs** [comment]
> Thank you for your great work.I modified your training code for joint sparse-view reconstruction and metal artifact reduction in CT images, but I found a problem while training the network.As you can see in the figure below, epochs keep looping between 13 and 14, I think training is in an endless cycle now. ![屏幕截图 2023-11-28 115112](https://private-user-images.githubusercontent.com/104410659/286115668-ff89c6d9-a7a0-4550-851e-e62262b8817a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNDQ1NzgsIm5iZiI6MTcwMTE0NDI3OCwicGF0aCI6Ii8xMDQ0MTA2NTkvMjg2MTE1NjY4LWZmODljNmQ5LWE3YTAtNDU1MC04NTFlLWU2MjI2MmI4ODE3YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyOFQwNDA0MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NGE3YjNhMjU4M2Y4ODU2NzAyZjQ5OWY0Y2U5NzdmZTQ4ZTdjYzhiMjYyZTA0NGE5MTBmYzBmNmJhZWRmZWM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.0EvHvKR1_w6pzxKhZGH4_2HmO76UaW8fxNpC__NHWWw)

Okay, my training code reload the saved best checkpoint and start over if it thinks backpropagation diverges, and then run scheduler one step to reduce the learning rate. It gets stuck in the endless cycle could mean the divergence check occurs every time it reloads the checkpoint, you may want to modify that or even turn that off entirely, or write your own scheduler rule.

Other things you can check to see if you changed the dataset to cause the data format not the same. The provided dataset sinogram ranges from 0 to 3.84, images range from 0 to 1.

And since the problem is different, you may want to try modifying the learning rate as well.

Hope this helps, and let me know if you discover what might cause the issue and have more questions.

**Comment by chenjun150** [comment]
Thank you for your great job.I want to change the size of input from 256 to 512,how can I do that?

**Comment by chrisdcs** [comment]
> Thank you for your great job.I want to change the size of input from 256 to 512,how can I do that?

256x256 is the CT image size, and 512 is the sinogram data size for each scan. If you have 64 views, the sinogram data size is 64x512. What do you mean by changing the size of input from 256 to 512?

**Comment by chenjun150** [comment]
Sorry for my misrepresentation of the issue. Since the size of the input CT image is 256*256，I want to verify the effect of 512*512, but I don't see any noticeable changes in the code to the size of the input image.

**Comment by chrisdcs** [comment]
> Sorry for my misrepresentation of the issue. Since the size of the input CT image is 256_256，I want to verify the effect of 512_512, but I don't see any noticeable changes in the code to the size of the input image.

You can add a line to resize the input image from 256x256 to 512x512 in the data loader. I did not change the size of the input images for any experiments. 

However, I don't recommend doing that because then you need to modify radon transform as well.

**Comment by chenjun150** [comment]
Thank you for your answer!
